File: 435-non-overlapping-intervals/non-overlapping-intervals.cpp
bool comp(vector<int>& i1, vector<int>& i2) {
    return i1[1] < i2[1];
}

class Solution {
public:
int eraseOverlapIntervals(vector<vector<int>>& intervals) {
    int count = 0;
    sort(intervals.begin(), intervals.end(), comp);
    int prevEnd = INT_MIN;
    for (const auto& interval : intervals) {
        int start = interval[0];
        int end = interval[1];

        if (start >= prevEnd) {
            prevEnd = end;
        } else {
            // Overlapping interval, increment count
            count++;
        }
    }
    return count;
}
};

File: 435-non-overlapping-intervals/non-overlapping-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
    function comp(i1, i2) {
        return i1[1] - i2[1];
    }

    intervals.sort(comp);
    let count = 0;
    let prevEnd = Number.MIN_SAFE_INTEGER;

    for (const interval of intervals) {
        const start = interval[0];
        const end = interval[1];

        if (start >= prevEnd) {
            prevEnd = end;
        } else {
            // Overlapping interval, increment count
            count++;
        }
    }

    return count;
};
